onnx/defs/schema.cc

```diff
@@ -94,7 +94,69 @@ OpSchemaRegistry* OpSchemaRegistry::Instance() {
   return &instance;
 }
 
-void OpSchema::Verify(const NodeProto& node, std::unordered_map<std::string, TypeProto*>* type_map) const {
+void OpSchema::CheckInputOutputType(const NodeProto& node, std::unordered_map<std::string, TypeProto*>* type_map) const {
+
+  std::unordered_map<std::string, DataType> type_constraints;
+  for (int in_idx = 0; in_idx < node.input_size() && in_idx < static_cast<int>(inputs_.size()); ++in_idx) {
+    const auto& all_supported_types = inputs_[in_idx].GetTypes();
+    const auto& type_iter = type_map->find(node.input(in_idx));
+    if (type_iter != type_map->end()) {
```

Comment:
I think we should fail_check if the type doesn't exist. (Same for outputs.)

Reply:
Now using inference context as suggested.
The context handles the nullptr situation by setting corresponding type as nullptr.